app/helpers/records_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module RecordsHelper
+  def record_kinds
+    {
+      login: Login.model_name.human(count: 0),
+      fail:  Fail.model_name.human(count: 0)
+    }
+  end
+
+  def record_user_name record
+    record.user.present? ? record.user.username : record.data['user_name']
+  end
+
+  def data_translate record
+    json = {}
+    record.data.each_with_object(json) do |(key, value), json_hash|
```

Comment:
```suggestion
    record.data.each_with_object({}) do |(key, value), json_hash|
```

Se puede poner derecho, pero el encare del tema quedó fantástico.

Reply:
genial. Tuve un ayudin de un amigo. Me mostro el metodo y esta bueno. Hay mucha magia en rails oculta jajaja